tests/framework/driver/assert_capture.py

```diff
@@ -0,0 +1,200 @@
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+from typing import Any
+from typing import List
+
+from tests.framework.interfaces.assert_forwarder import AssertForwarder
+
+
+class AssertCapture(AssertForwarder):
+    """
+    AssertForwarder implementation that wraps another AssertForwarder and
+    captures any exceptions thrown from it.
+    """
+
+    def __init__(self, basis: AssertForwarder):
+        """
+        Constructor
+
+        :param basis: AssertForwarder
+        """
+        self.basis: AssertForwarder = basis
+        self.asserts: List[AssertionError] = []
+
+    def get_asserts(self) -> List[AssertionError]:
+        """
+        :return: The list of asserts captured from the basis
+        """
+        return self.asserts
+
+    # pylint: disable=invalid-name
+    def assertEqual(self, first: Any, second: Any, msg: str = None):
+        """
+        Assert that the first is equal to the second
+
+        :param first: First comparison element
+        :param second: Second comparison element
+        :param msg: optional string message
+        """
+        try:
+            self.basis.assertEqual(first, second, msg)
+        except AssertionError as exception:
+            self.asserts.append(exception)
```

Comment:
All these methods are basically the same w/rt how the exception is caught from its basis and how it is sequestered for later use.